src/main/java/com/mixpanel/android/mpmetrics/DecideMessages.java

```diff
@@ -104,6 +104,13 @@ public synchronized void reportResults(List<Survey> newSurveys, List<InAppNotifi
             }
         }
 
+        // in the case we do not receive a new variant, this means the A/B test should be turned off
+        if(!hasNewVariants && newVariantsLength == 0) {
```

Comment:
@ryanseams You're right, we can only have one variant. Sorry! Can we just check `if (newVariantsLength == 0)` so we get rid of having a dependency on `hasNewVariants`?